### PR TITLE
chore(support-bundle): Added readme, fix api_url retrieval (for >= 6.9), fixed a regex for detect double digit version

### PR DESCRIPTION
**Pull request by victorbecerragit**
Added readme for support_bundle workaround for API_URL, fixed API_URL for Backend V6.9 and highers

### CODE REVIEW

**Comment by dark-vex** [comment]
I have removed the additional script and added the flags `-la` / `--local-api` that will automatically creates the kubernetes port forward.

While testing the script on a 6.10 and 6.11, I have spotted another bug. Support bundle script is unable to parse versions greater than 6.9 or 5.9, is going to return the error message:
```
We could not determine the backend version. Exiting.
```

I have changed the regex in order to fix it, this is what the previous regex was parsing:
Screenshot 2024-05-09 at 10 28 00
Screenshot 2024-05-09 at 10 28 36

**Review by dark-vex — DISMISSED** [review]

**Review by dark-vex — APPROVED** [review]

**Comment by dark-vex** [comment]
I have renamed also the PR title

**Review by cristof-ziel — DISMISSED** [review]

**Comment by dark-vex** [comment]
@patrickhargett @mbreitung could you please take a look at the PR and in case approve or suggesting any change if needed? 🙇

**Review by patrickhargett — DISMISSED** [review]
LGTM!

File: support_bundle/get_support_bundle.sh
Comment (mbreitung) [inline support_bundle/get_support_bundle.sh:156]: this will only pass for 6.x backend versions. It won't pick up 7.x. Is this expected?

**Review by mbreitung — APPROVED** [review]

**Review by cristof-ziel — APPROVED** [review]

**Comment by amoriroma** [comment]
Hi team, LY is waiting for this release. Can we proceed with this release? Thanks.

**Review by patrickhargett — DISMISSED** [review]

File: support_bundle/get_support_bundle.sh
Comment (mbreitung) [inline support_bundle/get_support_bundle.sh:155]: this seems overly complicated and not 100% accurate. This will also be true for versions 600.x for instance

Comment (mbreitung) [inline support_bundle/get_support_bundle.sh:155]:
```suggestion
        BACKEND_VERSION=$(kubectl ${CONTEXT_OPTS} ${KUBE_OPTS} get deployment sysdigcloud-api -ojsonpath='{.spec.template.spec.containers[0].image}' | awk -F: '{ print $2 }' | awk -F. '{ print $1 }' || true
```

File: support_bundle/get_support_bundle.sh
Comment (mbreitung) [inline support_bundle/get_support_bundle.sh:156]: Then this can be simplified to a non-regex
```suggestion
        if [[ "$BACKEND_VERSION" == "7" ]] || [[ "$BACKEND_VERSION" == "6" ]]; then
```

File: support_bundle/get_support_bundle.sh
Comment (mbreitung) [inline support_bundle/get_support_bundle.sh:184]:
```suggestion
        elif [[ "$BACKEND_VERSION" =~ ^(5|4|3)$ ]]; then
```

File: support_bundle/get_support_bundle.sh
Comment (mbreitung) [inline support_bundle/get_support_bundle.sh:233]: same comment here as above

Comment (mbreitung) [inline support_bundle/get_support_bundle.sh:233]:
```suggestion
        BACKEND_VERSION=$(kubectl ${CONTEXT_OPTS} ${KUBE_OPTS} get deployment sysdigcloud-api -ojsonpath='{.spec.template.spec.containers[0].image}' | awk -F: '{ print $2 }' | awk -F. '{ print $1 }') || true
```

File: support_bundle/get_support_bundle.sh
Comment (mbreitung) [inline support_bundle/get_support_bundle.sh:155]:
```suggestion
        BACKEND_VERSION=$(kubectl ${CONTEXT_OPTS} ${KUBE_OPTS} get deployment sysdigcloud-api -ojsonpath='{.spec.template.spec.containers[0].image}' | awk -F: '{ print $2 }') | awk -F. '{ print $1 }' || true
```

File: support_bundle/get_support_bundle.sh
Comment (mbreitung) [inline support_bundle/get_support_bundle.sh:234]:
```suggestion
        if [[ "$BACKEND_VERSION" =~ ^(7|6)$ ]]; then
```

File: support_bundle/get_support_bundle.sh
Comment (mbreitung) [inline support_bundle/get_support_bundle.sh:261]:
```suggestion
        elif [[ "$BACKEND_VERSION" =~ ^(5|4|3)$ ]]; then
```

File: support_bundle/get_support_bundle.sh
Comment (dark-vex) [reply support_bundle/get_support_bundle.sh]: This has been changed to:
```
BACKEND_VERSION=$(kubectl ${CONTEXT_OPTS} ${KUBE_OPTS} get deployment sysdigcloud-api -ojsonpath='{.spec.template.spec.containers[0].image}' | awk -F: '{ print $2 }' | awk -F. '{ print $1 }' || true
```
Marking the convo as resolved

**Review by mbreitung — APPROVED** [review]

**Review by dark-vex — APPROVED** [review]

**Comment by amoriroma** [comment]
Hi team, thank you for working on this. It worked fine in our lab, but LY got the same error in their environment.
```
% ./get_support_bundle.sh -a <MASKED_API_TOKEN> -n sysdigcloud

We could not determine the backend version. Exiting.
```
It does not seem to take into account the case where the Registry URL contains a port number, as in their environment.

- LY’s environment:

```
% kubectl -n sysdigcloud get deployment sysdigcloud-api -ojsonpath='{.spec.template.spec.containers[0].image}' | awk -F: '{ print $2 }' | awk -F. '{ print $1 }'

4443/sysdig/sysdig-api


% kubectl -n sysdigcloud get deployment sysdigcloud-api -ojsonpath='{.spec.template.spec.containers[0].image}'                      

cd.docker-registry.corp.yahoo.co.jp:4443/sysdig/sysdig-api:6.11.0.240423000256
```

- Our environment:

```
$ kubectl -n sysdig get deployment sysdigcloud-api -ojsonpath='{.spec.template.spec.containers[0].image}' | awk -F: '{ print $2 }' | awk -F. '{ print $1 }'
6

$ kubectl -n sysdig get deployment sysdigcloud-api -ojsonpath='{.spec.template.spec.containers[0].image}'
quay.io/sysdig/sysdig-api:6.11.0.240423000256
```
Can you make a further update to take into account cases where the Registry URL includes a port number, as in their environment? What about determining the version by customerVersion via api as described in this thread?
https://sysdigcloud.slack.com/archives/C03GJ1MN38Q/p1705660934469609?thread_ts=1705596392.908999&cid=C03GJ1MN38Q

Thanks,

**Pull request merged**
